docs/learn/smart-contract-developers/create-nft-collection-with-lsp7-tokenId.md

```diff
@@ -0,0 +1,956 @@
+- the [`LSP4DigitalAssetMetadata.sol`](https://github.com/lukso-network/lsp-smart-contracts/blob/develop/contracts/LSP4DigitalAssetMetadata/LSP4DigitalAssetMetadata.sol) contract
+- the [`_LSP4_SUPPORTED_STANDARDS_KEY`](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#supportedstandardslsp4digitalasset) constant
+- the [`_LSP4_SUPPORTED_STANDARDS_VALUE`](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#supportedstandardslsp4digitalasset) constant
+- the [`_LSP4_METADATA_KEY`](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata) constant
+- the [`_LSP8_REFERENCE_CONTRACT`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-8-IdentifiableDigitalAsset.md#lsp8referencecontract) constant
+
+```typescript title="contracts/LSP7Token.sol"
+import { LSP7DigitalAsset } from '@lukso/lsp-smart-contracts/contracts/LSP7DigitalAsset/LSP7DigitalAsset.sol';
+import { LSP4DigitalAssetMetadata } from '@lukso/lsp-smart-contracts/contracts/LSP4DigitalAssetMetadata/LSP4DigitalAssetMetadata.sol';
+
+import {
+  _LSP4_SUPPORTED_STANDARDS_KEY,
+  _LSP4_SUPPORTED_STANDARDS_VALUE,
+  _LSP4_METADATA_KEY
+} from '@lukso/lsp-smart-contracts/contracts/LSP4DigitalAssetMetadata/LSP4Constants.sol';
+import { _LSP8_REFERENCE_CONTRACT } from '@lukso/lsp-smart-contracts/contracts/LSP8IdentifiableDigitalAsset/LSP8Constants.sol';
+<!-- prettier-ignore-end -->
+```
+
+### Contract
+
+At deployment, we will need to set the following parameters:
+
+- `name_`: the name of the token
+- `symbol_`: the symbol of the token
+- `owner_`: the address that will be able to update the contract metadata (except the symbol, name and the LSP4TokenType)
+- `receiverOfInitialTokens_`: the address that will receive the minted tokens at deployment
+- `[lsp4TokenType\_](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-4-DigitalAsset-Metadata.md#lsp4tokentype): the token type of the token
+- [`isNonDivisible_`](../../contracts/contracts/LSP7DigitalAsset/LSP7DigitalAsset.md#decimals): whether the token is divisible or not
+- `totalSupply_`: the total supply of the token
+- `lsp4MetadataURI_`: the [LSP4 Metadata](../../standards//tokens/LSP4-Digital-Asset-Metadata.md#lsp4metadata)
+
+```typescript title="contracts/LSP8Token.sol"
+contract LSP7Token is LSP7DigitalAsset {
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        address newOwner_,
+        address receiverOfInitialTokens_,
+        uint256 lsp4TokenType_,
+        bool isNonDivisible_,
+        uint256 totalSupply_,
+        bytes memory lsp4MetadataURI_
+    )
+        LSP7DigitalAsset(
+            name_,
+            symbol_,
+            newOwner_,
+            lsp4TokenType_,
+            isNonDivisible_
+        )
+    {
+        // set SupportedStandards:LSP4DigitalAsset
+        LSP4DigitalAssetMetadata._setData(
```

Comment:
This line is not need, as it is already set by `LSP7DigitaAsset` via the parent `constructor` of `LSP4DigitalAssetMetadata`.

<img width="708" alt="image" src="https://github.com/lukso-network/docs/assets/31145285/de812291-67e0-4b1a-be62-8f7128f432cb">

See this link for reference to the code: https://github.com/lukso-network/lsp-smart-contracts/blob/def913ddabf007daad8cb486c9badb10a975af75/contracts/LSP4DigitalAssetMetadata/LSP4DigitalAssetMetadata.sol#L41-L45